src/providers/oauth2/types.ts

```diff
@@ -0,0 +1,37 @@
+import { OAuth2ProviderTypes } from "./mod.ts";
+
+export type ManifestOAuth2ProviderSchema = {
+  // provider_key: string;
+  provider_type: OAuth2ProviderTypes;
+  options: {
+    // deno-lint-ignore no-explicit-any
+    [key: string]: any;
+  };
+};
+
+export type OAuth2ProviderIdentitySchema = {
+  "url": string;
+  "account_identifier": string;
+  "headers"?: {
+    [key: string]: string;
+  };
+};
+
+export type OAuth2ProviderOptions = {
+  /** Client id for your provider */
+  "client_id": string;
+  /** Client secret for your provider */
+  "client_secret": string;
```

Comment:
We would actually want developers to reference an environment variable for the secret rather than having it be hardcoded in the code. I know we have it in `FunctionContext` but is it possible to expose env in the manifest as well?